src/Avalonia.Labs.Gif/GifImage.cs

```diff
@@ -181,6 +182,10 @@ private void InitializeGif()
         if (Source is Stream s)
         {
             stream = s;
+            if (stream.Position != 0 && stream.CanSeek)
```

Comment:
This is technically incorrect and is a misuse of Stream API. GIF data could be located somewhere in a custom container.